File: 백준/Silver/11724. 연결 요소의 개수/연결 요소의 개수.cc
#include <bits/stdc++.h>
using namespace std;

int n, m;
vector<int> adj_list[1002];
bool visited[1002];

void bfs(int x) {
    visited[x] = true;

    queue<int> q;
    q.push(x);

    while(!q.empty()) {
        auto cur = q.front();
        q.pop();

        for(int data : adj_list[cur]) {
            if(visited[data]) {
                continue;
            }
            q.push(data);
            visited[data] = true;
        }
    }
}

int main(void)
{
    ios::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);

    cin >> n >> m;

    while(m--) {
        int u, v;
        cin >> u >> v;
        adj_list[u].push_back(v);
        adj_list[v].push_back(u);
    }

    int area = 0;
    for(int i=1; i<=n; i++) {
        if(!visited[i]) {
            area++;
            bfs(i);
        }
    }

    cout << area;

    return 0;
}

File: 백준/Silver/11724. 연결 요소의 개수/연결 요소의 개수.java
import java.util.*;

public class Main {
    static int n, m;
    static List<Integer>[] adjList;
    static boolean[] visited;

    public static void bfs(int x) {
        Queue<Integer> q = new LinkedList<>();
        visited[x] = true;
        q.offer(x);

        while (!q.isEmpty()) {
            int cur = q.poll();
            for (int neighbor : adjList[cur]) {
                if (!visited[neighbor]) {
                    visited[neighbor] = true;
                    q.offer(neighbor);
                }
            }
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        n = sc.nextInt();
        m = sc.nextInt();

        adjList = new ArrayList[n + 1];
        visited = new boolean[n + 1];

        for (int i = 1; i <= n; i++) {
            adjList[i] = new ArrayList<>();
        }

        for (int i = 0; i < m; i++) {
            int u = sc.nextInt();
            int v = sc.nextInt();
            adjList[u].add(v);
            adjList[v].add(u);
        }

        int area = 0;
        for (int i = 1; i <= n; i++) {
            if (!visited[i]) {
                area++;
                bfs(i);
            }
        }

        System.out.println(area);
        sc.close();
    }
}
